internal/supported_types/embed.go

```diff
@@ -0,0 +1,27 @@
+package supported_types
+
+import (
+	"embed"
+
+	"gopkg.in/yaml.v2"
+)
+
+var awsSupported = make(map[string]interface{})
+
+//go:embed supported_instance_types.yml
+var f embed.FS
+
+func init() {
+	it, err := f.ReadFile("supported_instance_types.yml")
+	if err != nil {
+		panic(err)
+	}
+
+	if err := yaml.Unmarshal(it, &supported); err != nil {
+		panic(err)
+	}
+
+	for _, i := range supported.Aws.Ec2 {
+		awsSupported[i] = nil
+	}
```

Comment:
ugh? ^_^

Reply:
the map I was referring to :D
we can change it to a slice with binary search

Reply:
ah, right, got it now... `i` is very missleading, I thought it is the index and thus got confused.

Reply:
Lists are more human-friendly then maps in YAML, the alternative was to sort out the slices and use bisearch, which Adi obviously did not like. I thought it would be more coding fun but I am not doing this feature, sadly :)

Reply:
I am fine with this, no need to do the search. By the way binary search is not the only option, there is plethora of other more complicated ones :)